dask_expr/_collection.py

```diff
@@ -400,6 +400,25 @@ def __getitem__(self, other):
             other = other.copy()
         return new_collection(self.expr.__getitem__(other))
 
+    def __repr__(self):
+        data = self._repr_data().to_string(max_rows=5)
+        _str_fmt = """Dask {klass} Structure:
+{data}
+Dask Name: {name}, {layers}
+Expr={expr}"""
+        if len(self.columns) == 0:
+            data = data.partition("\n")[-1].replace("Index", "Divisions")
+            _str_fmt = f"Empty {_str_fmt}"
+        layers = len({k for (k, *_) in self.dask.keys()})
```

Comment:
This triggers a graph materialization. This must not happen. We should revert this and figure out something else

cc @milesgranger